hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteOperatorCoordinator.java

```diff
@@ -503,10 +509,29 @@ private boolean commitInstant(String instant, long checkpointId) {
       sendCommitAckEvents(checkpointId);
       return false;
     }
+    setMinEventTime();
     doCommit(instant, writeResults);
+    resetMinEventTime();
     return true;
   }
 
+  public void setMinEventTime() {
+    if (commitEventTimeEnable) {
+      LOG.info("receive event time for current commit: {} ", Arrays.stream(eventBuffer).collect(Collectors.toMap(WriteMetadataEvent::getTaskID, WriteMetadataEvent::getMaxEventTime)));
+      this.minEventTime = Arrays.stream(eventBuffer)
+        .filter(Objects::nonNull)
+        .map(WriteMetadataEvent::getMaxEventTime)
+        .filter(maxEventTime -> maxEventTime > 0)
+        .min(Comparator.naturalOrder())
+        .map(aLong -> Math.min(aLong, this.minEventTime)).orElse(Long.MAX_VALUE);
+      LOG.info("[setMinEventTime] minEventTime: {} ", this.minEventTime);
+    }
+  }
+
+  public void resetMinEventTime() {
```

Comment:
can remove this method?

Reply:
no, we need reset min event time for each commit